File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-confirmation-processes-5-4/MaturityNotification.cpp
// MaturityNotification.cpp 
#include "MaturityNotification.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

MaturityNotification::MaturityNotification(TiXmlNode* xmlNode)
: CorrectableRequestMessage(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //optionExpiryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* optionExpiryNode = xmlNode->FirstChildElement("optionExpiry");

   if(optionExpiryNode){optionExpiryIsNull_ = false;}
   else{optionExpiryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- optionExpiryNode , address : " << optionExpiryNode << std::endl;
   #endif
   if(optionExpiryNode)
   {
      if(optionExpiryNode->Attribute("href") || optionExpiryNode->Attribute("id"))
      {
          if(optionExpiryNode->Attribute("id"))
          {
             optionExpiryIDRef_ = optionExpiryNode->Attribute("id");
             optionExpiry_ = boost::shared_ptr<OptionExpiryBase>(new OptionExpiryBase(optionExpiryNode));
             optionExpiry_->setID(optionExpiryIDRef_);
             IDManager::instance().setID(optionExpiryIDRef_,optionExpiry_);
          }
          else if(optionExpiryNode->Attribute("href")) { optionExpiryIDRef_ = optionExpiryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { optionExpiry_ = boost::shared_ptr<OptionExpiryBase>(new OptionExpiryBase(optionExpiryNode));}
   }

   //tradeMaturityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* tradeMaturityNode = xmlNode->FirstChildElement("tradeMaturity");

   if(tradeMaturityNode){tradeMaturityIsNull_ = false;}
   else{tradeMaturityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- tradeMaturityNode , address : " << tradeMaturityNode << std::endl;
   #endif
   if(tradeMaturityNode)
   {
      if(tradeMaturityNode->Attribute("href") || tradeMaturityNode->Attribute("id"))
      {
          if(tradeMaturityNode->Attribute("id"))
          {
             tradeMaturityIDRef_ = tradeMaturityNode->Attribute("id");
             tradeMaturity_ = boost::shared_ptr<TradeMaturity>(new TradeMaturity(tradeMaturityNode));
             tradeMaturity_->setID(tradeMaturityIDRef_);
             IDManager::instance().setID(tradeMaturityIDRef_,tradeMaturity_);
          }
          else if(tradeMaturityNode->Attribute("href")) { tradeMaturityIDRef_ = tradeMaturityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { tradeMaturity_ = boost::shared_ptr<TradeMaturity>(new TradeMaturity(tradeMaturityNode));}
   }

   //partyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyNode = xmlNode->FirstChildElement("party");

   if(partyNode){partyIsNull_ = false;}
   else{partyIsNull_ = true;}

   if(partyNode)
   {
      for(partyNode; partyNode; partyNode = partyNode->NextSiblingElement("party")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
          #endif
          if(partyNode->Attribute("href") || partyNode->Attribute("id"))
          {
              if(partyNode->Attribute("id"))
              {
                  partyIDRef_ = partyNode->Attribute("id");
                  party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));
                  party_.back()->setID(partyIDRef_);
                  IDManager::instance().setID(partyIDRef_, party_.back());
              }
              else if(partyNode->Attribute("href")) { partyIDRef_ = partyNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { party_.push_back(boost::shared_ptr<Party>(new Party(partyNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyNode , address : " << partyNode << std::endl;
       #endif
   }

   //accountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* accountNode = xmlNode->FirstChildElement("account");

   if(accountNode){accountIsNull_ = false;}
   else{accountIsNull_ = true;}

   if(accountNode)
   {
      for(accountNode; accountNode; accountNode = accountNode->NextSiblingElement("account")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
          #endif
          if(accountNode->Attribute("href") || accountNode->Attribute("id"))
          {
              if(accountNode->Attribute("id"))
              {
                  accountIDRef_ = accountNode->Attribute("id");
                  account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));
                  account_.back()->setID(accountIDRef_);
                  IDManager::instance().setID(accountIDRef_, account_.back());
              }
              else if(accountNode->Attribute("href")) { accountIDRef_ = accountNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { account_.push_back(boost::shared_ptr<Account>(new Account(accountNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- accountNode , address : " << accountNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<OptionExpiryBase> MaturityNotification::getOptionExpiry()
{
   if(!this->optionExpiryIsNull_){
        if(optionExpiryIDRef_ != ""){
             return boost::shared_static_cast<OptionExpiryBase>(IDManager::instance().getID(optionExpiryIDRef_));
        }else{
             return this->optionExpiry_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<OptionExpiryBase>();
   }
}
boost::shared_ptr<TradeMaturity> MaturityNotification::getTradeMaturity()
{
   if(!this->tradeMaturityIsNull_){
        if(tradeMaturityIDRef_ != ""){
             return boost::shared_static_cast<TradeMaturity>(IDManager::instance().getID(tradeMaturityIDRef_));
        }else{
             return this->tradeMaturity_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<TradeMaturity>();
   }
}
std::vector<boost::shared_ptr<Party>> MaturityNotification::getParty()
{
   if(!this->partyIsNull_){
        if(partyIDRef_ != ""){
             return this->party_;
        }else{
             return this->party_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Party>>();
   }
}
std::vector<boost::shared_ptr<Account>> MaturityNotification::getAccount()
{
   if(!this->accountIsNull_){
        if(accountIDRef_ != ""){
             return this->account_;
        }else{
             return this->account_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Account>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-confirmation-processes-5-4/MaturityNotification.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class MaturityNotification : CorrectableRequestMessage
    {
        public MaturityNotification(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode optionExpiryNode = xmlNode.SelectSingleNode("optionExpiry");
            
            if (optionExpiryNode != null)
            {
                if (optionExpiryNode.Attributes["href"] != null || optionExpiryNode.Attributes["id"] != null) 
                {
                    if (optionExpiryNode.Attributes["id"] != null) 
                    {
                        optionExpiryIDRef_ = optionExpiryNode.Attributes["id"].Value;
                        OptionExpiryBase ob = new OptionExpiryBase(optionExpiryNode);
                        IDManager.SetID(optionExpiryIDRef_, ob);
                    }
                    else if (optionExpiryNode.Attributes["href"] != null)
                    {
                        optionExpiryIDRef_ = optionExpiryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        optionExpiry_ = new OptionExpiryBase(optionExpiryNode);
                    }
                }
                else
                {
                    optionExpiry_ = new OptionExpiryBase(optionExpiryNode);
                }
            }
            
        
            XmlNode tradeMaturityNode = xmlNode.SelectSingleNode("tradeMaturity");
            
            if (tradeMaturityNode != null)
            {
                if (tradeMaturityNode.Attributes["href"] != null || tradeMaturityNode.Attributes["id"] != null) 
                {
                    if (tradeMaturityNode.Attributes["id"] != null) 
                    {
                        tradeMaturityIDRef_ = tradeMaturityNode.Attributes["id"].Value;
                        TradeMaturity ob = new TradeMaturity(tradeMaturityNode);
                        IDManager.SetID(tradeMaturityIDRef_, ob);
                    }
                    else if (tradeMaturityNode.Attributes["href"] != null)
                    {
                        tradeMaturityIDRef_ = tradeMaturityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        tradeMaturity_ = new TradeMaturity(tradeMaturityNode);
                    }
                }
                else
                {
                    tradeMaturity_ = new TradeMaturity(tradeMaturityNode);
                }
            }
            
        
            XmlNodeList partyNodeList = xmlNode.SelectNodes("party");
            
            if (partyNodeList != null)
            {
                this.party_ = new List<Party>();
                foreach (XmlNode item in partyNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            partyIDRef_ = item.Attributes["id"].Value;
                            party_.Add(new Party(item));
                            IDManager.SetID(partyIDRef_, party_[party_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            partyIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        party_.Add(new Party(item));
                        }
                    }
                    else
                    {
                        party_.Add(new Party(item));
                    }
                }
            }
            
        
            XmlNodeList accountNodeList = xmlNode.SelectNodes("account");
            
            if (accountNodeList != null)
            {
                this.account_ = new List<Account>();
                foreach (XmlNode item in accountNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            accountIDRef_ = item.Attributes["id"].Value;
                            account_.Add(new Account(item));
                            IDManager.SetID(accountIDRef_, account_[account_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            accountIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        account_.Add(new Account(item));
                        }
                    }
                    else
                    {
                        account_.Add(new Account(item));
                    }
                }
            }
            
        
        }
        
    
        #region OptionExpiry_
        private OptionExpiryBase optionExpiry_;
        public OptionExpiryBase OptionExpiry_
        {
            get
            {
                if (this.optionExpiry_ != null)
                {
                    return this.optionExpiry_; 
                }
                else if (this.optionExpiryIDRef_ != null)
                {
                    optionExpiry_ = IDManager.getID(optionExpiryIDRef_) as OptionExpiryBase;
                    return this.optionExpiry_; 
                }
                else
                {
                      return this.optionExpiry_; 
                }
            }
            set
            {
                if (this.optionExpiry_ != value)
                {
                    this.optionExpiry_ = value;
                }
            }
        }
        #endregion
        
        public string optionExpiryIDRef_ { get; set; }
        #region TradeMaturity_
        private TradeMaturity tradeMaturity_;
        public TradeMaturity TradeMaturity_
        {
            get
            {
                if (this.tradeMaturity_ != null)
                {
                    return this.tradeMaturity_; 
                }
                else if (this.tradeMaturityIDRef_ != null)
                {
                    tradeMaturity_ = IDManager.getID(tradeMaturityIDRef_) as TradeMaturity;
                    return this.tradeMaturity_; 
                }
                else
                {
                      return this.tradeMaturity_; 
                }
            }
            set
            {
                if (this.tradeMaturity_ != value)
                {
                    this.tradeMaturity_ = value;
                }
            }
        }
        #endregion
        
        public string tradeMaturityIDRef_ { get; set; }
        #region Party_
        private List<Party> party_;
        public List<Party> Party_
        {
            get
            {
                if (this.party_ != null)
                {
                    return this.party_; 
                }
                else if (this.partyIDRef_ != null)
                {
                    return this.party_; 
                }
                else
                {
                      return this.party_; 
                }
            }
            set
            {
                if (this.party_ != value)
                {
                    this.party_ = value;
                }
            }
        }
        #endregion
        
        public string partyIDRef_ { get; set; }
        #region Account_
        private List<Account> account_;
        public List<Account> Account_
        {
            get
            {
                if (this.account_ != null)
                {
                    return this.account_; 
                }
                else if (this.accountIDRef_ != null)
                {
                    return this.account_; 
                }
                else
                {
                      return this.account_; 
                }
            }
            set
            {
                if (this.account_ != value)
                {
                    this.account_ = value;
                }
            }
        }
        #endregion
        
        public string accountIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

